core/domain/topic_services.py

```diff
@@ -88,6 +89,48 @@ def get_all_skill_ids_assigned_to_some_topic():
     return skill_ids
 
 
+def get_all_skill_ids_assigned_to_some_topic_with_topic_details():
+    """Returns the ids of all the skills that are linked to some topics
+       and name of the classroom to which the topic is assigned to.
+
+    Returns:
+        dict. A dict with skill_id as key and topic and classroom
+        name as values.
+    """
+    skill_ids_with_topic_details = {}
+
+    all_topic_models = topic_models.TopicModel.get_all()
+    all_topic_ids = [topic_model.id for topic_model in all_topic_models]
+    all_topics = topic_fetchers.get_topics_by_ids(all_topic_ids)
+
+    for topic in all_topics:
+        for skill_id in topic.get_all_skill_ids():
+            classroom_name = get_classroom_name_for_topic_with_id(topic.id)
+            skill_ids_with_topic_details[skill_id] = {
```

Comment:
No -- should be returning a domain object, always. (But given its current usage, I suggest just inlining it.)

Reply:
I have made changes to this, but not sure if that's what you said. So please check this once.